### PR TITLE
Fix lambda expansion

### LINKED ISSUE #6130: Lambdas are no longer inlined
That's because `is_inlinable_formula()` has a typo in it:

```diff
modified   R/across.R
@@ -612,7 +612,8 @@ as_across_fn_call <- function(fn, var, env, mask) {
 # bug.
 is_inlinable_formula <- function(x, mask) {
   if (is_formula(x, lhs = FALSE, scoped = TRUE)) {
-    identical(x, mask) || !env_inherits(f_env(x), mask)
+    env <- f_env(x)
+    identical(env, mask) || !env_inherits(env, mask)
   } else {
     FALSE
   }
```

We need a test for expansions.

Also I wonder if we could bind `.` and `.x` to `cur_column()` when evaluating expansions of `across()`? See https://github.com/tidyverse/glue/issues/242. The reprex worked on dev dplyr because of the expansion regression.

### PR DESCRIPTION
**Pull request by lionel-**
Closes #6130

### CODE REVIEW

File: R/across.R [hunk 1]
```diff
@@ -571,7 +571,11 @@ expand_across <- function(quo) {
   }
 
   names(expressions) <- names
-  expressions
+  new_expanded_quosures(expressions)
+}
+
+new_expanded_quosures <- function(x) {
```
Comment (lionel-) [inline R/across.R:577]: I originally extracted this ctor for unit tests. I didn't end up using it in the end, but it's probably a bit cleaner this way.

Comment (romainfrancois) [reply R/across.R:577]: yeah definitely. Thanks.

**Review by romainfrancois — CHANGES_REQUESTED** [review]
Perhaps we should also have similar tests for `expand_if_across()`. 

It looks like there a cleanup issue: 

```
[lambda-expansion*] 311.71 MB > cur_group_id()
Error in `cur_group_id()`:
! Must be used inside dplyr verbs.
Run `rlang::last_error()` to see where the error occurred.
[lambda-expansion*] 313.20 MB > test_that("expand_across() expands lambdas", {
+   quo <- quo(across(c(cyl, am), ~ identity(.x)))
+   quo <- new_dplyr_quosure(
+     quo,
+     name_given = "",
+     name_auto = "across()",
+     is_named = FALSE,
+     index = 1
+   )
+ 
+   mask <- DataMask$new(mtcars, current_env(), "mutate", call("caller"))
+   local_mask(mask)
+ 
+   expect_equal(
+     map(expand_across(quo), quo_get_expr),
+     exprs(
+       cyl = identity(cyl),
+       am = identity(am)
+     )
+   )
+ })
Test passed 😀
[lambda-expansion*] 315.11 MB > 
[lambda-expansion*] 315.11 MB > cur_group_id()
[1] 0
```

This causes a later test to fail: 

```
x | 1      18 | context [0.7s]                                                                                                                                                                                                                     
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Error (test-context.R:92:3): give useful error messages when not applicable
`{ ... }` threw an unexpected error.
Message: `cur_group_id()` did not throw the expected error.
Class:   expectation_failure/expectation/error/condition
Backtrace:
 1. testthat::expect_error(cur_group_id())
 2. testthat:::expect_condition_matching(...)
```

File: tests/testthat/test-across.R [hunk 2]
```diff
@@ -858,6 +858,28 @@ test_that("across() predicates operate on whole data", {
...
+  quo <- new_dplyr_quosure(
+    quo,
+    name_given = "",
+    name_auto = "across()",
+    is_named = FALSE,
+    index = 1
+  )
+
+  mask <- DataMask$new(mtcars, current_env(), "mutate", call("caller"))
+  local_mask(mask)
```
Comment (romainfrancois) [inline tests/testthat/test-across.R:872]:
```suggestion
```

I believe this is not needed `local_mask()` is called inside `DataMask$new()`

Comment (lionel-) [reply tests/testthat/test-across.R:872]: oh. It's surprising that a ctor has side effects.

**Review by romainfrancois — APPROVED** [review]

**Pull request merged**